pkg/apis/core/validation/validation_test.go

```diff
@@ -9952,159 +9909,149 @@ func TestValidatePodSpec(t *testing.T) {
+			podtest.SetVolumes(podtest.MakeEmptyVolume(("vol"))),
+			podtest.SetSecurityContext(core.PodSecurityContext{
 				HostIPC: true,
-			},
-			Volumes:       []core.Volume{{Name: "vol", VolumeSource: core.VolumeSource{EmptyDir: &core.EmptyDirVolumeSource{}}}},
-			Containers:    []core.Container{{Name: "ctr", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: "File"}},
-			RestartPolicy: core.RestartPolicyAlways,
-			DNSPolicy:     core.DNSClusterFirst,
-		},
-		"populate HostPID": {
-			SecurityContext: &core.PodSecurityContext{
+			}),
+		).Spec,
+		"populate HostPID": podtest.MakePod("",
+			podtest.SetContainers(podtest.MakeContainer("ctr")),
+			podtest.SetVolumes(podtest.MakeEmptyVolume(("vol"))),
+			podtest.SetSecurityContext(core.PodSecurityContext{
 				HostPID: true,
-			},
-			Volumes:       []core.Volume{{Name: "vol", VolumeSource: core.VolumeSource{EmptyDir: &core.EmptyDirVolumeSource{}}}},
-			Containers:    []core.Container{{Name: "ctr", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: "File"}},
-			RestartPolicy: core.RestartPolicyAlways,
-			DNSPolicy:     core.DNSClusterFirst,
-		},
-		"populate Affinity": {
-			Volumes:       []core.Volume{{Name: "vol", VolumeSource: core.VolumeSource{EmptyDir: &core.EmptyDirVolumeSource{}}}},
-			Containers:    []core.Container{{Name: "ctr", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: "File"}},
-			RestartPolicy: core.RestartPolicyAlways,
-			DNSPolicy:     core.DNSClusterFirst,
-		},
-		"populate HostAliases": {
-			HostAliases:   []core.HostAlias{{IP: "12.34.56.78", Hostnames: []string{"host1", "host2"}}},
-			Volumes:       []core.Volume{{Name: "vol", VolumeSource: core.VolumeSource{EmptyDir: &core.EmptyDirVolumeSource{}}}},
-			Containers:    []core.Container{{Name: "ctr", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: "File"}},
-			RestartPolicy: core.RestartPolicyAlways,
-			DNSPolicy:     core.DNSClusterFirst,
-		},
-		"populate HostAliases with `foo.bar` hostnames": {
-			HostAliases:   []core.HostAlias{{IP: "12.34.56.78", Hostnames: []string{"host1.foo", "host2.bar"}}},
-			Volumes:       []core.Volume{{Name: "vol", VolumeSource: core.VolumeSource{EmptyDir: &core.EmptyDirVolumeSource{}}}},
-			Containers:    []core.Container{{Name: "ctr", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: "File"}},
-			RestartPolicy: core.RestartPolicyAlways,
-			DNSPolicy:     core.DNSClusterFirst,
-		},
-		"populate HostAliases with HostNetwork": {
-			HostAliases: []core.HostAlias{{IP: "12.34.56.78", Hostnames: []string{"host1.foo", "host2.bar"}}},
-			Containers:  []core.Container{{Name: "ctr", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: "File"}},
-			SecurityContext: &core.PodSecurityContext{
+			}),
+		).Spec,
+		"populate Affinity": podtest.MakePod("",
+			podtest.SetContainers(podtest.MakeContainer("ctr")),
```

Comment:
I think the goal of this pattern for tests is to make every testcase spell out as LITTLE as possible.  Calling validation on `podtest.MakePod("foo")` should succeed.  Thenceforth, everything you specify in a testcase is germane to that specific testcase.  All these calls to `podtest.SetContainers(podtest.MakeContainer("ctr"))` can go away - they are just noise and are not related to the testcases (except when they are).

Reply:
Likewise, almost none of these test cases need a volume

Reply:
So my question is whether `podtest.MakePod("foo")` creates a valid pod or not. Currently it does not since containers are required. Now I could do the reverse and create a default container which can be replaced which is what I think you are getting at?

Reply:
Let me know what you think of these changes. I removed what made sense to me.